homeassistant/components/light/hue.py

```diff
@@ -302,7 +302,7 @@ def __init__(self, light_id, info, bridge, update_lights,
     def unique_id(self):
         """Return the ID of this Hue light."""
         return "{}.{}".format(
-            self.__class__, self.info.get('uniqueid', self.name))
+            self.__class__, self.info.get('uniqueid', self.light_id))
```

Comment:
The unique ID in Home Assistant is never exposed. It is only used when passing devices to `add_devices` to filter out duplicates. It will be fine to add the type to it.